src/nrniv/nonlinz.cpp

```diff
@@ -245,46 +228,21 @@ NonLinImpRep::NonLinImpRep() {
     if (neq_ == 0) {
         return;
     }
-    m_ = cmplx_spCreate(neq_, 1, &err);
-    assert(err == spOKAY);
+    m_.resize(neq_, neq_);
     pv_.resize(neq_);
     pvdot_.resize(neq_);
-    v_index_ = new int[n_v_];
-    rv_ = new double[neq_ + 1];
-    rv_ += 1;
-    jv_ = new double[neq_ + 1];
-    jv_ += 1;
-    diag_ = new double*[neq_];
```

Comment:
so I take it `diag_` went away because `m_` now holds the full matrix including the diagonal and Eigen handles everything internally?

Reply:
`diag_` have been removed because this was a pointer access directly inside the matrix structure.
diagonals numbers have always been in the matrix (how could we solve otherwise).
`Eigen::SparseMatrix` is a bit like a `int*` that can be reallocated if needed.
The pointer of number inside `Eigen` are not stable for `Sparse`.
In `sparse13` the number were stored as a linked list and so stable.

Reply:
ok, makes sense